bin/cherry-pick.mjs

```diff
@@ -0,0 +1,416 @@
+/**
+ * External dependencies
+ */
+import fetch from 'node-fetch';
+import readline from 'readline';
+
+import { spawnSync } from 'node:child_process';
+
+const LABEL = "Backport to WP Beta/RC";
+const BRANCH = getCurrentBranch();
+const GITHUB_CLI_AVAILABLE = spawnSync( 'gh', ['auth', 'status'] )
+	?.stderr
+	?.toString()
+	.includes( '✓ Logged in to github.com as' );
+
+const AUTO_PROPAGATE_RESULTS_TO_GITHUB = GITHUB_CLI_AVAILABLE;
+
+/**
+ * The main function of this script. It:
+ * * Confirms with the developer the current branch aligns with the expectations
+ * * Gets local branches in sync with the remote ones
+ * * Requests the list of PRs to cherry-pick from GitHub API (closed, label=`Backport to WP Beta/RC`)
+ * * Runs `git cherry-pick {commitHash}` for each PR
+ * * It keeps track of the failed cherry-picks and then retries them
+ * * Retrying keeps going as long as at least one cherry-pick succeeds
+ * * Pushes the local branch to `origin`
+ * * (optional) Uses the [`gh` console utility](https://cli.github.com/) to comment on the remote PRs and remove the labels
+ * * Reports the results
+ */
+async function main() {
+	console.log( `You are on branch "${BRANCH}".` );
+	console.log( `This script will:` );
+	console.log( `• Cherry-pick the merged PRs labeled as "${LABEL}" to this branch` );
+	console.log( `• Push this branch` );
+	console.log( `• Comment on each PR` );
+	console.log( `• Remove the label from each PR` );
+	console.log( `` );
+	await promptDoYouWantToProceed();
+
+	if ( !GITHUB_CLI_AVAILABLE ) {
+		await reportGhUnavailable();
+	}
+
+	console.log( `$ git pull origin ${ BRANCH } --rebase...` );
+	cli( 'git', ['pull', 'origin', BRANCH, '--rebase'], true );
+
+	console.log( `$ git fetch origin trunk...` );
+	cli( 'git', ['fetch', 'origin', 'trunk'], true );
+
+	const PRs = await fetchPRs();
+	console.log( 'Trying to cherry-pick one by one...' );
+	const [successes, failures] = cherryPickAll( PRs );
+	console.log( 'Cherry-picking finished!' );
+
+	reportSummaryNextSteps( successes, failures );
+
+	if ( successes.length ) {
+		if ( AUTO_PROPAGATE_RESULTS_TO_GITHUB ) {
```

Comment:
Not necessarily, if `GITHUB_CLI_AVAILABLE` is `false` this if will still get executed and it might go to the `else` branch. `successes.length` are all about cherry-picks, they're unrelated to `GITHUB_CLI_AVAILABLE`.

Reply:
Okay, I misread the code. We don't bail out when `GITHUB_CLI_AVAILABLE` is false, we only show a notice 👍🏻